programs/stake/src/stake_state.rs

```diff
@@ -418,8 +418,8 @@ impl Stake {
         let mut points = 0;
         let mut new_credits_observed = self.credits_observed;
 
-        for (epoch, final_epoch_credits, initial_epoch_credits) in
-            new_vote_state.epoch_credits().iter().copied()
+        if let Some((epoch, final_epoch_credits, initial_epoch_credits)) =
+            new_vote_state.epoch_credits().last().copied()
```

Comment:
this is known that this ins't enough.

We need explicitly check whether the last epoch_credits is target_epoch for this inflation reward distribution. Otherwise,  we could be tricked by stale vote accounts.

Reply:
now this is pushed.